packages/steps/src/BuildWorkflow.ts

```diff
@@ -16,8 +16,22 @@ export class BuildWorkflow {
   }
 
   public async executeAsync(): Promise<void> {
+    let maybeError: Error | null = null;
+    let hasAnyPreviousStepsFailed = false;
     for (const step of this.buildSteps) {
-      await step.executeAsync();
+      if (step.shouldExecuteStep(hasAnyPreviousStepsFailed)) {
+        try {
+          await step.executeAsync();
+        } catch (err: any) {
+          maybeError = err;
```

Comment:
Should we error with the first or last error? Why?

Reply:
I believe that the first error is usually the true one. The rest of the errors will happen in the cleanup steps which should `always()` be run.

Reply:
Exactly. If the "cleanup step" fails, wouldn't we accidentally reassign `maybeError` and forget the first `err`?

Reply:
Good catch, sorry, I will fix it